src/datacollector/Program.cs

```diff
@@ -62,6 +70,19 @@ private static void Run(string[] args)
             var argsDictionary = CommandLineArgumentsHelper.GetArgumentsDictionary(args);
             var requestHandler = DataCollectionRequestHandler.Create(new SocketCommunicationManager(), new MessageSink());
 
+            // Attach to exit of parent process
+            string optionValue;
+            var parentProcessId = argsDictionary.TryGetValue(ParentProcessArgument, out optionValue) ? int.Parse(optionValue) : 0;
```

Comment:
Parsing Int values can be moved to CoreUtilities as it is used by both testhost and datacollector.